src/lib/cmd.rs

```diff
@@ -7,16 +7,27 @@ fn remote_cmd(args: &mut Vec<&str>, output: &mut File) -> Result<(), String> {
     // TODO(azhng): let's finger cross this works.
     let output_file = output.try_clone().unwrap();
     let error_file = output.try_clone().unwrap();
-    let cmd = Command::new("ssh")
+    let cmd_process = Command::new("ssh")
         .arg(config::SEXXI_REMOTE_HOST)
-        .args(args)
+        .args(args.clone())
         .stdout(Stdio::from(output_file))
         .stderr(Stdio::from(error_file))
-        .output()
-        .expect("Ok");
+        .spawn();
+
+    match cmd_process {
+        Ok(child) => {
+            println!("Server process with id {} is running command `{:?}`", child.id(), args);
+
+            let result = child.wait_with_output().expect("Ok");
+            if !result.status.success() {
+                return Err(format!("remote command failed: {}", result.status));
+            }
+        },
+        Err(e) => {
+            println!("Command `{:?}` failed, server process didn't start: {}", args, e);
+            std::process::exit(1);
```

Comment:
does this line terminate the entire server ?

Reply:
From what I understand it terminates the server process

Reply:
I think the ideal thing to do here is wrap the `Err` with more detailed message and bubble it up to the caller. It is caller's responsibility to handle the error in this case instead of directly terminating the entire server process. It is also undesirable to kill the server process here since we don't have infra to restart the server (e.g. k8s or systemd).

Reply:
Yeah I think bubling it up will be nice, also probably should send the error message about this to github

Reply:
Killing proccesses should usually not be the error handling mecahnism, especially if not restart is in place

Reply:
This may need to be a different PR